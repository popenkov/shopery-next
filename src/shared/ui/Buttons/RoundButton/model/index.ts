export { type TRoundButtonTheme } from './button.types';
