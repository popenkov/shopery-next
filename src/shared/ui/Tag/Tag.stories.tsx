import type { Meta, StoryObj } from '@storybook/react';
import { Logo, LogoSize } from './Tag';

const meta = {
  title: 'shared/Logo',
  component: Logo,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof Logo>;

export default meta;
type Story = StoryObj<typeof meta>;

export const DefaultLogo: Story = {
  args: {
    size: LogoSize.DEFAULT,
  },
};

export const SmallLogo: Story = {
  args: {
    size: LogoSize.SMALL,
  },
};
