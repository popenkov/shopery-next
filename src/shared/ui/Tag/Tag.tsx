import { DetailedHTMLProps, FC, HTMLAttributes, memo } from 'react';
import cn from 'classnames';
import cls from './Tag.module.scss';

export enum TagVariant {
  RED = 'red',
  BLUE = 'blue',
  ORANGE = 'orange',
  BLACK = 'black',
}

interface TagProps
  extends DetailedHTMLProps<HTMLAttributes<HTMLSpanElement>, HTMLSpanElement> {
  content: string;
  variant: TagVariant;
}

export const Tag: FC<TagProps> = memo(({ content, variant }) => {
  return (
    <span
      className={cn(cls.tag, {
        [cls[variant]]: true,
      })}
    >
      {content}
    </span>
  );
});
