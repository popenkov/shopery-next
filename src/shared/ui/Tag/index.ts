export { Tag } from './ui';
export { type TTag } from './model';
