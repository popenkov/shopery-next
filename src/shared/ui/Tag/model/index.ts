export { type TTag, type TTagVariant } from './types';
