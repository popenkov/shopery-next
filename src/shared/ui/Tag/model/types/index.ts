export { type TTag, type TTagVariant } from './tag.types';
