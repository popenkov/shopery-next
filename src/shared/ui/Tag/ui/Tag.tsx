import { DetailedHTMLProps, FC, HTMLAttributes, ReactNode, memo } from 'react';

import cn from 'classnames';

import cls from './Tag.module.scss';
import { Text } from '../../Text';

type TagVariant =
    | 'sale'
    | 'not-available'
    | 'best-sale'
    | 'new'
    | 'out-of-stock'
    | 'in-stock';

interface TagProps
    extends DetailedHTMLProps<
        HTMLAttributes<HTMLSpanElement>,
        HTMLSpanElement
    > {
    children: ReactNode;
    variant: TagVariant;
}

export const Tag: FC<TagProps> = memo(({ children, variant }) => {
    return (
        <Text
            variant="body_s"
            className={cn(cls.tag, {
                [cls[variant]]: true,
            })}
            as="span"
        >
            {children}
        </Text>
    );
});

Tag.displayName = 'Tag';
