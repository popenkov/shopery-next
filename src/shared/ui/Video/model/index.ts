export { type TVideo } from './video.types';
