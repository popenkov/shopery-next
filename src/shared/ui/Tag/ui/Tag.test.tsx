import React from 'react';
import { render } from '@testing-library/react';

import { TTagVariant } from '../model';

import { Tag } from './Tag';

describe('Tag component', () => {
  it('renders correctly', () => {
    const { container } = render(<Tag variant="sale">Sale</Tag>);
    expect(container).toMatchSnapshot();
  });

  it('renders correct class names', () => {
    const { container } = render(<Tag variant="sale">Sale</Tag>);
    expect(container.querySelector('.tag')).toBeInTheDocument();
    expect(container.querySelector('.sale')).toBeInTheDocument();
  });

  it('renders correct text', () => {
    const { getByText } = render(<Tag variant="sale">Sale</Tag>);
    expect(getByText('Sale')).toBeInTheDocument();
  });

  it('renders correct variant classes', () => {
    const variants = ['sale', 'not-available', 'best-sale', 'new', 'out-of-stock', 'in-stock'];
    variants.forEach((variant) => {
      const { container } = render(<Tag variant={variant as TTagVariant}>{variant}</Tag>);
      expect(container.querySelector(`.${variant}`)).toBeInTheDocument();
    });
  });

  it('renders as span element', () => {
    const { container } = render(<Tag variant="sale">Sale</Tag>);
    expect(container.querySelector('span')).toBeInTheDocument();
  });
});
