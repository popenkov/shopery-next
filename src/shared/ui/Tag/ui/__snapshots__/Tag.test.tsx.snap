// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component renders correctly 1`] = `
<div>
  <span
    class="text tag sale primary body_s left regular"
  >
    Sale
  </span>
</div>
`;
